dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskInstance.java

```diff
@@ -306,12 +306,16 @@ public void init(String host, Date startTime, String executePath) {
     }
 
     public DependentParameters getDependency() {
-        if (this.dependency == null) {
+        if (this.dependency == null && TASK_TYPE_DEPENDENT.equalsIgnoreCase(this.taskType)) {
             Map<String, Object> taskParamsMap =
                     JSONUtils.parseObject(this.getTaskParams(), new TypeReference<Map<String, Object>>() {
                     });
+            Object dependence = taskParamsMap.get(Constants.DEPENDENCE);
+            if (dependence == null) {
+                return null;
+            }
```

Comment:
Returning null in here may cause NPE in the downstream stage. We should find a way to avoid this. cc @ruanwenjun

Reply:
Without this check, JSONUtils.parseObject function will catch exception and return null, return value is also null

Reply:
And I think other stage must check it for avoid NPE, because can't guarantee it's not null in the downstream stage even though return not null value in here.